rust/lance/src/io/commit.rs

```diff
@@ -182,11 +182,11 @@ pub(crate) async fn migrate_fragments(
     let dataset = Arc::new(dataset.clone());
     let new_fragments = futures::stream::iter(fragments)
         .map(|fragment| async {
-            let physical_rows = if fragment.physical_rows == 0 {
-                let file_fragment = FileFragment::new(dataset.clone(), fragment.clone());
-                Either::Left(async move { file_fragment.count_rows().await })
```

Comment:
I'm investigating still, but I think using `count_rows()` instead of `physical_rows()` might have been a serious bug.

Reply:
This is indeed a bug: https://github.com/lancedb/lance/issues/1531

Reply:
Moving to solve this in a separate PR.